File: java-advent-of-code/src/test/java/advent/twenty_sixteen/Day08TwoFactorAuthenticationTest.java
package advent.twenty_sixteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day08TwoFactorAuthenticationTest {
    private static final String REAL = //
            ".##..####.###..#..#.###..####.###....##.###...###.\n" + //
            "#..#.#....#..#.#..#.#..#....#.#..#....#.#..#.#....\n" + //
            "#..#.###..###..#..#.#..#...#..###.....#.#..#.#....\n" + //
            "####.#....#..#.#..#.###...#...#..#....#.###...##..\n" + //
            "#..#.#....#..#.#..#.#....#....#..#.#..#.#.......#.\n" + //
            "#..#.#....###...##..#....####.###...##..#....###..\n";

    @Test
    public void checkReal() {
        DailyProblem<Integer, String> dp = new Day08TwoFactorAuthentication("/twenty_sixteen/Day08-TwoFactorAuthentication-input.txt", 6, 50);

        assertEquals(123, dp.getPart1Answer().intValue());
        assertEquals(REAL, dp.getPart2Answer());
    }

    private static final String EXAMPLE_1 = //
            ".#..#.#\n" + //
            "#.#....\n" + //
            ".#.....\n";

    @Test
    public void checkExample1() {
        DailyProblem<Integer, String> dp = new Day08TwoFactorAuthentication("/twenty_sixteen/Day08-TwoFactorAuthentication-example#1.txt", 3 , 7);

        assertEquals(6, dp.getPart1Answer().intValue());
        assertEquals(EXAMPLE_1, dp.getPart2Answer());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_sixteen/Day08TwoFactorAuthenticationTest.scala
package advent.twenty_sixteen

import org.scalatest.{FlatSpec, Matchers}

class Day08TwoFactorAuthenticationTest  extends FlatSpec with Matchers {
  "2016 Day 08 - Input" should "calculate answers" in {
    val dp = new Day08TwoFactorAuthentication("/twenty_sixteen/Day08-TwoFactorAuthentication-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day08TwoFactorAuthentication("/twenty_sixteen/Day08-TwoFactorAuthentication-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
